core-processor/src/main/java/io/micronaut/inject/annotation/internal/KotlinDeprecatedTransformer.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2017-2022 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.inject.annotation.internal;
+
+import io.micronaut.core.annotation.AnnotationValue;
+import io.micronaut.inject.annotation.NamedAnnotationTransformer;
+import io.micronaut.inject.visitor.VisitorContext;
+
+import java.lang.annotation.Annotation;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * Allows treating the Kotlin deprecated annotation as the Java one.
+ *
+ * @since 4.0.0
+ */
+public final class KotlinDeprecatedTransformer implements NamedAnnotationTransformer {
```

Comment:
Why do we need to remap the deprecated annotation?

Reply:
because we have some logic that excludes configuration properties based on the annotation https://github.com/micronaut-projects/micronaut-core/blob/3cac50ae10d6ab84a64e6840fa53ea128bd833c8/core-processor/src/main/java/io/micronaut/inject/processing/ConfigurationReaderBeanElementCreator.java#L287